include/boost/url/pct_encoding.hpp

```diff
@@ -182,121 +245,63 @@ validate_pct_encoding(
-    assert( result == "Program Files" );
+    char *dest = new char[MAX_LENGTH];
+    error_code ec;
+    std::size_t decoded_size = pct_decode( dest, dest + MAX_LENGTH,
+            "Program%20Files", ec, pct_decode_opts{});
+
+    assert( ! ec.failed() );
+    assert( decoded_size == 13 );
+    assert( strncmp( "Program Files", dest, decoded_size ) == 0 );
     @endcode
 
     @par Exception Safety
-    Throws on invalid input.
-    Calls to allocate may throw.
-
-    @return A `std::basic_string` holding
-    the decoded result. If the default
-    allocator is used, the return type is
-    `std::string`.
-
-    @param s The string to decode.
-
-    @param opt The options for decoding. If
-    this parameter is omitted, the default
-    options will be used.
-
-    @param cs An optionally specified
-    character set to use. If this parameter
-    is omitted, all characters are considered
-    unreserved.
-
-    @param a An optional allocator the returned
-    string will use. If this parameter is omitted,
-    the default allocator is used.
-
-    @throws std::invalid_argument invalid input.
-
-    @par Specification
-    @li <a href="https://datatracker.ietf.org/doc/html/rfc3986#section-2.1"
-        >2.1. Percent-Encoding (rfc3986)</a>
-
-    @see
-        @ref pct_decode_bytes_unchecked,
-        @ref pct_decode_opts,
-        @ref pct_decode_to_value.
-        @ref pct_decode_unchecked.
-        @ref validate_pct_encoding.
-*/
-template<
-    class CharSet = reserved_chars_t,
-    class Allocator = std::allocator<char> >
-std::basic_string<char,
-    std::char_traits<char>,
-        Allocator>
-pct_decode(
-    string_view s,
-    pct_decode_opts const& opt = {},
-    CharSet const& cs = {},
-    Allocator const& a = {});
-
-/** Return a string with percent-decoding applied.
-
-    This function applies percent-decoding to
-    the given percent-encoded string, by
-    converting escape sequences into their
-    character equivalent.
-    The result is returned as a @ref const_string
-    using the optionally specified allocator.
+    Throws nothing.
 
-    @par Example
-    @code
-    const_string result = pct_decode_to_value( "Program%20Files", pct_decode_opts{}, pchars );
-    assert( result.compare( "Program Files" ) == 0 );
-    @endcode
+    @return The number of bytes written to
+    the destination buffer, which does not
+    include any null termination.
 
-    @par Exception Safety
-    Throws on invalid input.
-    Calls to allocate may throw.
+    @param dest A pointer to the
+    beginning of the output buffer.
 
-    @return A `std::basic_string` holding
-    the decoded result. If the default
-    allocator is used, the return type is
-    `std::string`.
+    @param end A pointer to one past the end
+    of the output buffer.
 
     @param s The string to decode.
 
+    @param ec Set to the error, if any
+    occurred. If the destination buffer
+    is too small to hold the result, `ec`
+    is set to @ref error::no_space.
```

Comment:
You shouldn't go down to the level of detail of specifying the error for trivial things, unless the error is something that happens on a regular basis (for example: asio's `error::eof` when the socket is closed). Because if we change it, then we have to change the doc. And most users won't see this error anyway. And the ones that do see the error, they can look it up and they will know what it means.